beast/tests/test_regresscheck.py

```diff
@@ -849,6 +850,29 @@ def test_star_type_probability_no_Av(self):
         # compare to new table
         compare_tables(expected_star_prob, Table(star_prob))
 
+    def test_calc_depth_from_completeness(self):
+        """
+        Test for calculate_depth.py
+        """
+        # calculate depth for 50% and 75% completeness
+        depth = calc_depth_from_completeness.calc_depth(
+            self.seds_fname_cache,
+            self.noise_fname_cache,
+            completeness_value=[0.5, 0.75],
+            vega_mag=True,
+        )
+        # expected results
+        expected_dict = {
+            "HST_WFC3_F275W": [25.000309202589012, 24.80610510139205],
+            "HST_WFC3_F336W": [24.65974845352875, 24.338061586936263],
+            "HST_ACS_WFC_F475W": [np.nan, np.nan],
+            "HST_ACS_WFC_F814W": [np.nan, 24.368742437736692],
+            "HST_WFC3_F110W": [np.nan, np.nan],
```

Comment:
Why are these `nan` values?  Is there something wrong with the ASTs in the phat_small example?  Mainly curious.

Reply:
As far as I've figured out, there are two reasons for nans.  One is if the ASTs don't cover the whole flux range of the grid, so there aren't statistics for faint sources.  The other is if the observations are so deep that even the faintest sources are recovered at some non-zero rate.

Reply:
Here's an example for one of the METAL fields, using completenesses of 0.2, 0.25, 0.3:
```
{'HST_WFC3_F225W': [42.48340001978902, 36.85778870609744, 36.54311413627377],
'HST_WFC3_F275W': [nan, nan, 37.442233109245905],
'HST_WFC3_F336W': [nan, 38.19925138344242, 37.2154373318946],
'HST_WFC3_F475W': [nan, nan, 33.83419535967331],
'HST_WFC3_F814W': [28.871186766091327, 28.685084890012547, 28.526476367397507],
'HST_WFC3_F110W': [26.802812634408944, 26.66268095677285, 26.538354275870823],
'HST_WFC3_F160W': [25.54145798073774, 25.387076912540735, 25.252994542836568]}
```
What I think this is saying (and please correct me if I'm wrong) is that for the faintest blue sources, they're bright enough in another filter to still be recovered ~25% of the time.

Reply:
Ok.  Thanks.  This just emphasizes that it would be great to get a better set of observations/ASTs for the regression tests and example run.  But that is a topic for another PR.